src/lib/middleware/rateLimiting.ts

```diff
@@ -0,0 +1,53 @@
+import config from "../../config"
+import { RateLimiterMemory, RateLimiterRedis } from "rate-limiter-flexible"
+
+const {
+  OPENREDIS_URL,
+  REQUEST_EXPIRES,
+  REQUEST_LIMIT,
+  REQUEST_PER_INSTANCE_FALLBACK,
+  ENABLE_RATE_LIMITING,
+} = config
+
+export const rateLimiterMiddlewareFactory = redisClient => {
+  /**
+   * Used as a per process limiter if for whatever reason
+   * redis becomes unavailable
+   */
+  const rateLimiterMemory = new RateLimiterMemory({
+    points: REQUEST_PER_INSTANCE_FALLBACK,
+    duration: REQUEST_EXPIRES,
+  })
```

Comment:
So this is only for when Redis is unavailable during startup of the process, does that mean that the below Redis based limiter will fallback to in-memory tracking when Redis becomes unavailable after the process already started?

Reply:
Yes. It’s used if redis isn’t available on startup or as the insurance limited if it becomes unavailable later.